### PR TITLE
Use common poller

### PR DESCRIPTION
**Pull request by aarsilv**
🎟️  **Ticket:** [FF-1471 - Have Node SDK use updated Common JS Module](https://linear.app/eppo/issue/FF-1471/have-node-sdk-use-updated-common-js-module)

## Motivation and Context
So that our client-side JavaScript SDK can benefit from the polling improvements done to this SDK, we're moving `poller` to `js-client-sdk-common`. 

👀 I recommend checking out that repository's [PR #35 - Common configuration requestor/poller ](https://github.com/Eppo-exp/js-client-sdk-common/pull/35) as part of reviewing this one.

## Description
Removes `poller` and instead adjusts this SDK to leverage it within `js-client-sdk-common`.
This greatly simplifies this SDK 📈 

## How has this been tested?
Leveraged the previously added unit tests in `index.spec.ts`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,10 +29,8 @@
   },
   "homepage": "https://github.com/Eppo-exp/node-server-sdk#readme",
   "dependencies": {
-    "axios": "^1.6.0",
-    "http-status-codes": "^2.2.0",
-    "lru-cache": "^10.0.1",
-    "@eppo/js-client-sdk-common": "^2.0.0"
+    "@eppo/js-client-sdk-common": "file:../js-client-sdk-common",
```
Comment (aarsilv) [inline package.json:32]: We will publish `js-client-sdk-common` before merging this PR, and replace this with its latest release version.

File: src/index.spec.ts [hunk 2]
```diff
@@ -68,11 +71,34 @@ describe('EppoClient E2E test', () => {
...
+        'cancelAnimationFrame',
+        'requestIdleCallback',
+        'cancelIdleCallback',
+        'setImmediate',
+        'clearImmediate',
+        'setInterval',
+        'clearInterval',
+      ],
+    });
+  });
```
Comment (aarsilv) [inline src/index.spec.ts:93]: :old-man-yells-at-jest:

File: src/index.ts [hunk 3]
```diff
@@ -81,43 +75,30 @@ let clientInstance: IEppoClient;
...
   clientInstance.setLogger(config.assignmentLogger);
 
   // default to LRU cache with 50_000 entries.
   // we estimate this will use no more than 10 MB of memory
   // and should be appropriate for most server-side use cases.
   clientInstance.useLRUInMemoryAssignmentCache(50_000);
 
-  await poller.start();
+  // Fetch configurations (which will also start regular polling per requestConfiguration)
+  await clientInstance.fetchFlagConfigurations();
```
Comment (leoromanovsky) [inline src/index.ts:101]: nice

**Review by leoromanovsky — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (sameerank) [reply package.json:32]: Ah hence the failing lint and type checks

**Review by sameerank — APPROVED** [review]

**Pull request merged**